src/test/regress/upgrade/citus_upgrade_test.py

```diff
@@ -0,0 +1,154 @@
+        with utils.cd('/'):
+            ps = subprocess.Popen(('tar', 'tf', abs_tar_path), stdout=subprocess.PIPE)
+            output = subprocess.check_output(('xargs', '-d', '\n', 'rm', '-v'), stdin=ps.stdout)
+            ps.wait()            
+
+def remove_citus_artifacts(citus_version, pg_version):
+    with utils.cd('/'):
+        ps = subprocess.Popen(('tar', 'tf', './install-pg{}-citus{}.tar'.format(pg_version, citus_version)), stdout=subprocess.PIPE)
+        output = subprocess.check_output(('xargs', '-d', '\n', 'rm', '-v'), stdin=ps.stdout)
+        ps.wait()
+        
+
+def run_alter_citus_mixed_mode(pg_path):
+    for port in NODE_PORTS.values():
+        if port == WORKER1PORT:
+            continue
+        utils.psql(pg_path, port, "ALTER EXTENSION citus UPDATE;")
+
+
+def run_alter_citus(pg_path):
+    for port in NODE_PORTS.values():
+        utils.psql(pg_path, port, "ALTER EXTENSION citus UPDATE;")
+
+
+def restart_database(pg_path, abs_data_path, node_name):
+    command = [
+        os.path.join(pg_path, 'pg_ctl'), 'restart',
+        '--pgdata', abs_data_path,
+        '-U', USER,
+        '-o', '-p {}'.format(NODE_PORTS[node_name]),
+        '--log', os.path.join(abs_data_path, 'logfile_' + node_name)
+    ]
+    subprocess.call(command)
+
+
+def restart_databases_mixed_mode(pg_path, rel_data_path):
+    for node_name in NODE_NAMES:
+        if NODE_PORTS[node_name] == WORKER1PORT:
+            continue
+        abs_data_path = os.path.abspath(os.path.join(rel_data_path, node_name))
+        restart_database(
+            pg_path=pg_path, abs_data_path=abs_data_path, node_name=node_name)
+
+
+def restart_databases(pg_path, rel_data_path):
+    for node_name in NODE_NAMES:
+        abs_data_path = os.path.abspath(os.path.join(rel_data_path, node_name))
+        restart_database(
+            pg_path=pg_path, abs_data_path=abs_data_path, node_name=node_name)
+
+
+def run_citus_upgrade_test(config, mixed_mode):
+    install_citus(config.citus_version, config.pg_version)
+    initialize_temp_dir(config.temp_dir)
+    initialize_citus_cluster(
+        config.bindir, config.datadir, config.settings)
+
+    verify_initial_version(config)
+    run_test_on_coordinator(config, BEFORE_CITUS_UPGRADE_COORD_SCHEDULE)
+    remove_citus_artifacts(config.citus_version, config.pg_version)
+    install_citus_master(config.pg_version)
+
+    if mixed_mode:
+        restart_databases_mixed_mode(config.bindir, config.datadir)
+        run_alter_citus_mixed_mode(config.bindir)
+        verify_upgrade_mixed_mode(config)
+    else:
+        restart_databases(config.bindir, config.datadir)
+        run_alter_citus(config.bindir)
+        verify_upgrade(config)
+
+    run_test_on_coordinator(config, AFTER_CITUS_UPGRADE_COORD_SCHEDULE)
+    remove_citus_master_artifacts(config.pg_version)
+
+def main(config):
+    run_citus_upgrade_test(config=config, mixed_mode=False)
```

Comment:
We run `run_citus_upgrade_test` for mixed_mode = `False` and `True`. Therefore we cant put the body of function here. However currently I get an error when I enable the mixed mode. You can see the error in the PR description. I will look into the error so that we can test `mixed_mode` enabled as well.